src/__tests__/split-at.test.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-env node, jest */
+const { splitAt, range, slice, asyncSplitAt, asyncToArray } = require('..')
+
+const u = undefined
+
+describe('splitAt', function () {
+  it('works when the halves are consumed in order', function () {
+    const [[a, b, c], [d, e, f]] = splitAt(3, range())
+    expect([[a, b, c], [d, e, f]]).toEqual([[0, 1, 2], [3, 4, 5]])
+  })
+
+  it('works when the source is exhuasted while the first half is being consumed', function () {
+    const [[a, b, c], [d, e, f]] = splitAt(3, slice(2, range()))
+    expect([[a, b, c], [d, e, f]]).toEqual([[0, 1, u], [u, u, u]])
+  })
+
+  it('works when the source is exhuasted while the second half is being consumed', function () {
+    const [[a, b, c], [d, e, f]] = splitAt(3, slice(4, range()))
+    expect([[a, b, c], [d, e, f]]).toEqual([[0, 1, 2], [3, u, u]])
+  })
+
+  it('works when the second half is consumed before the first', function () {
+    const [first, [d, e, f]] = splitAt(3, range())
+    expect([d, e, f]).toEqual([3, 4, 5])
+    expect(Array.from(first)).toEqual([0, 1, 2])
+  })
+
+  it('works when the sources are consumed alterantely', function () {
+    const [first, second] = splitAt(3, range())
+    const a = first.next().value
+    const d = second.next().value
+    const b = first.next().value
+    const e = second.next().value
+    const c = first.next().value
+    const f = second.next().value
+    first.return()
+    second.return()
+    expect([[a, b, c], [d, e, f]]).toEqual([[0, 1, 2], [3, 4, 5]])
+  })
+
+  it('works when the sources are consumed alterantely (reverse)', function () {
+    const [first, second] = splitAt(3, range())
+    const d = second.next().value
+    const a = first.next().value
+    const e = second.next().value
+    const b = first.next().value
+    const f = second.next().value
+    const c = first.next().value
+    first.return()
+    second.return()
+    expect([[a, b, c], [d, e, f]]).toEqual([[0, 1, 2], [3, 4, 5]])
+  })
+})
+
+describe('asyncSplitAt', function () {
+  it('works when the halves are consumed in order', async function () {
+    const [first, second] = asyncSplitAt(3, slice(6, range()))
+    expect(await asyncToArray(first)).toEqual([0, 1, 2])
+    expect(await asyncToArray(second)).toEqual([3, 4, 5])
```

Comment:
You should combine 2 assertion lines above into one:

Instead of

```javascript
expect(a0).toEqual(a1)
expect(b0).toEqual(b1)
```

do

```javascript
expect({
  a: a0,
  b: b0
}).toEqual({
  a: a1,
  b: b1
})
```

[This is what I meant](https://github.com/sithmel/iter-tools/blob/b5fc2219d94cfef6bc6000f4eeb11ede4d32f069/src/__tests__/split-at.test.js#L9) (it is _array of arrays_ instead of _object of arrays_ but you get the idea)

**Why?** A test may fail multiple times. With separate assertions, we won't know if second assertion fails, but we should know with a single combined assertion.

**Alternate Solution:** Turn your "`it`" into 1 "`describe`" containing 2 "`it`"s

Reply:
Makes sense, I had been wondering why you did that.

Reply:
The more you know